contracts/lbp/LBPManagerFactory.sol

```diff
@@ -0,0 +1,141 @@
+            "LBPManagerFactory: mastercopy can not be zero"
+        );
+        require(
+            _masterCopy != address(this),
+            "LBPManagerFactory: mastercopy can not be the same as LBPManagerFactory"
+        );
+        lbpManagerMasterCopy = _masterCopy;
+    }
+
+    /**
+     * @dev                             Set Balancers LBP Factory contract as basis for deploying LBPs.
+     * @param _LBPFactory               The address of Balancers LBP factory.
+     */
+    function setLBPFactory(address _LBPFactory) external onlyOwner {
+        require(
+            _LBPFactory != address(0),
+            "LBPManagerFactory: LBPFactory can not be zero"
+        );
+        require(
+            _LBPFactory != address(this),
+            "LBPManagerFactory: LBPFactory can not be the same as LBPManagerFactory"
+        );
+        LBPFactory = _LBPFactory;
+    }
+
+    /**
+     * @dev                             Deploy and initialize LBPManager.
+     * @param _admin                    The address of the admin of the LBPManager.
+     * @param _beneficiary              The address that receives the _primeDaoFeePercentage.
+     * @param _name                     Name of the LBP.
+     * @param _symbol                   Symbol of the LBP.
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _amounts                  Sorted array to match the _tokenList, containing two parameters:
+                                            - The amounts of project token to be added as liquidity to the LBP.
+                                            - The amounts of funding token to be added as liquidity to the LBP.
+     * @param _weights                  Sorted array to match the _tokenList, containing two parametes:
+                                            - The start weight for the project token in the LBP.
+                                            - The start weight for the funding token in the LBP.
+     * @param _startTimeEndtime         Array containing two parameters:
+                                            - Start time for the LBP.
+                                            - End time for the LBP.
+     * @param _endWeights               Sorted array to match the _tokenList, containing two parametes:
+                                            - The end weight for the project token in the LBP.
+                                            - The end weight for the funding token in the LBP.
+     * @param _swapFeePercentage        Percentage of fee paid for every swap in the LBP.
+     * @param _primeDaoFeePercentage    Percentage of fee paid to PrimeDao for providing the service of the LBP Manager.
+     */
+    function deployLBPUsingManager(
```

Comment:
why do we have to specify that the deployment happens with `Manager`, can we just name it `deployLBP`?

Reply:
In a later PR, I changed it to `deployLBPManger`. Would that also be okay or change it to `deployLBP`?